### PR TITLE
Add YouTube video caption tool and related stories/tests

### PR DESCRIPTION
**Pull request by elkernel128**
## Description

This pull request introduces the YouTube Video Caption Tool to the KaibanJS project. This tool extracts video captions from YouTube videos using the YouTube Data API and `youtube-captions-scraper`.

## Changes

- Added the `YouTubeVideoCaption` class in `index.js`.
- Implemented functionality to fetch and process captions for a given YouTube video ID.
- Defined the input schema using Zod for validation.

## Requirements

1. **YouTube API Token**
   - You need an API token to authenticate requests to the YouTube Data API.
   - This token should be passed when creating an instance of the `YouTubeVideoCaption` class.

2. **Input Parameters**
   - The tool requires a single input parameter:
     - `videoId`: A string representing the ID of the YouTube video from which you want to extract captions.

## Example Usage

```jsx
import { YouTubeVideoCaption } from './index.js';

// Create an instance of the tool with the required API token
const youtubeCaptionTool = new YouTubeVideoCaption({ token: 'YOUR_YOUTUBE_API_TOKEN' });

// Define the input parameter
const input = {
  videoId: 'VIDEO_ID_HERE',
};

// Call the tool to extract captions
youtubeCaptionTool._call(input).then((captions) => {
  console.log(captions);
}).catch((error) => {
  console.error(error);
});
```

## Checklist

- [x] Added `YouTubeVideoCaption` class.
- [x] Implemented caption extraction functionality.
- [x] Defined input schema using Zod.
- [x] Added unit tests for the new tool.
- [x] Verified that all tests pass.

## How to Test

1. Clone the repository and navigate to the project directory.
2. Install dependencies: `npm install`
3. Run tests: `npm run test`
4. Verify that the `YouTubeVideoCaption` tool works as expected by following the usage instructions in the updated `README.md` file.

---

### CODE REVIEW

File: packages/tools/src/youtube-video-caption/youtubeVideoCaption.stories.jsx
Comment (darielnoel) [inline packages/tools/src/youtube-video-caption/youtubeVideoCaption.stories.jsx:1]: LEt's use for the name of the file tool.stories.jsx

File: packages/tools/src/youtube-video-caption/youtubeVideoCaption.test.js
Comment (darielnoel) [inline packages/tools/src/youtube-video-caption/youtubeVideoCaption.test.js:1]: Let's use for the name of the file

LEt's use for the name of the file tool.test.js

**Review by darielnoel — COMMENTED** [review]
Check minor details

**Comment by darielnoel** [comment]
Don't forget to add the Tool basic doc to the Readme

https://github.com/kaiban-ai/KaibanJS/blob/main/packages/tools/README.md

**Comment by anthonydevs17** [comment]
#### Guide: Obtain YouTube API Authorization Token Without Custom Credentials

This guide will help you get a YouTube API authorization token for testing purposes using Google’s OAuth 2.0 Playground without requiring a Client ID and Client Secret.

##### **Step 1: Access OAuth 2.0 Playground**

- Go to the [OAuth 2.0 Playground](https://developers.google.com/oauthplayground/).
- This tool allows you to interact with Google APIs and generate authorization tokens.

##### **Step 2: Select YouTube API Scopes**

- Look for the **YouTube Data API scopes** in Step 1 of the Playground.
- Choose the appropriate scopes depending on the API actions you want to test:
  - **`https://www.googleapis.com/auth/youtube`**: Full access to manage YouTube account.
  - **`https://www.googleapis.com/auth/youtube.readonly`**: Read-only access to YouTube account.
  - **`https://www.googleapis.com/auth/youtube.force-ssl`**: Allows uploading, managing, and retrieving video content securely, including captions.

##### **Step 3: Authorize the APIs**

- Click the **"Authorize APIs"** button.
- A Google sign-in page will appear:
  - Sign in with your Google account.
  - Grant the requested permissions.

##### **Step 4: Exchange Authorization Code for Tokens**

- After successful authorization, go to Step 2 of the Playground.
- Click the **"Exchange authorization code for tokens"** button.
- The Playground will generate:
  - **Access Token:** Use this token to make API requests.
  - (Optional) **Refresh Token:** If available, use this to get a new access token when the current one expires.

##### **Step 5: Test API Requests**

- Use Step 3 of the Playground to test API requests directly:
  1. Enter the YouTube API endpoint (e.g., `https://www.googleapis.com/youtube/v3/videos`).
  2. Add required parameters and headers.
  3. Use the generated access token for authentication.

##### **Limitations**

- **Short-lived Access Token:** The token typically expires in 1 hour.
- **Default Credentials Restrictions:** Some advanced API features may not be accessible.
- **No Persistent Refresh Token:** A new authorization process might be needed for extended testing.

##### **Recommendation for Advanced Testing**

For more comprehensive testing, create your own project in the [Google Cloud Console](https://console.cloud.google.com/) and obtain a **Client ID** and **Client Secret** to fully unlock the YouTube API features.

**Comment by anthonydevs17** [comment]
Implement with different approach soon...

**Pull request closed without merging**